make-util.js

```diff
@@ -1850,6 +1850,98 @@ var processGeneratedTasks = function(baseConfigToolPath, taskList, makeOptions,
+ */
+var mergeBuildConfigIntoBaseTasks = function(buildConfig) {
+    var makeOptionsPath = path.join(__dirname, 'make-options.json');
+    var makeOptions = fileToJson(makeOptionsPath);
+    if (!makeOptions) fail("makeOptions is not defined");
+    const AllTasksToMerge = makeOptions[buildConfig];
+    const TasksfailedToMerge = [];
+
+    const match = buildConfig.match(/^([a-zA-Z]+[0-9]+)_\d+(_\d+)?$/);
+    var surfixBuildConfig = match ? match[1] : buildConfig;
+
+    if (AllTasksToMerge && Array.isArray(AllTasksToMerge)) {
+        AllTasksToMerge.forEach(taskName => {
+            var generatedTaskPath = path.join(__dirname, '_generated', `${taskName}_${surfixBuildConfig}`);
+            var generatedDefaultTaskPath = path.join(__dirname, '_generated', taskName)
+            var versionmapFilePath = path.join(__dirname, '_generated', `${taskName}.versionmap.txt`);
+            var baseTaskPath = path.join(__dirname, 'Tasks', taskName);
+            var buildConfigTaskPath = path.join(baseTaskPath, '_buildConfigs', surfixBuildConfig);
+
+            if (!fs.existsSync(generatedTaskPath) || !fs.statSync(generatedTaskPath).isDirectory() || !fs.existsSync(versionmapFilePath)) {
+                console.log(`Invalid generated task path ${generatedTaskPath} or invalid ${taskName}.versionmap.txt file ${versionmapFilePath}\n`);
+                TasksfailedToMerge.push(taskName);
+            } else {
+                banner(`Merging ${generatedTaskPath} into base task...`);
+
+                // Copy generated task to base task, delete generated files
+                cp('-rf', generatedTaskPath + "/*", baseTaskPath);
+                console.log(`Copied ${generatedTaskPath} to ${baseTaskPath}`);
+                rm("-rf", buildConfigTaskPath);
+                console.log(`Deleted ${buildConfigTaskPath} folder`);
+                rm("-rf", generatedTaskPath);
+                console.log(`Deleted ${generatedTaskPath} folder`);
+                rm("-rf", generatedDefaultTaskPath);
+                console.log(`Deleted ${generatedDefaultTaskPath} folder`);
+
+                // Update versionmap.txt file
+                var versionmapFile = fs.readFileSync(versionmapFilePath, { encoding: 'utf-8' });
+                const lines = versionmapFile.split('\n');
+                var buildConfigVersion = null;
+                for (let i = lines.length - 1; i >= 0; i--) {
+                    if (!lines[i]) continue;
+                    const [name, version] = lines[i].split('|');
+                    if (name.startsWith(surfixBuildConfig)) {
+                        buildConfigVersion = version;
+                        lines.splice(i, 1);
+                        i++;
+                    } else if (name === "Default" && buildConfigVersion != null) {
+                        lines[i] = `Default|${buildConfigVersion}`;
+                    }
+                }
+                const updatedContent = lines.join('\n');
```

Comment:
[something to check] unless there's magic going on, this may end up with the wrong line ending for windows.  if we change this, we shouldn't that the line endings are \r\n either